Framework/Core/src/LifetimeHelpers.cxx

```diff
@@ -42,26 +42,28 @@ size_t getCurrentTime()
 
 ExpirationHandler::Creator LifetimeHelpers::dataDrivenCreation()
 {
-  return [](TimesliceIndex&) -> void {
-    return;
+  return [](TimesliceIndex&) -> TimesliceSlot {
+    return { TimesliceSlot::INVALID };
```

Comment:
Maybe we should change the API so that we allow the creator to create multiple time slots? or at least have it being wrapped in `std::optional`.

Reply:
I am not sure what would the gain of using `std::optional` aside from `TimesliceSlot::INVALID`. The multiple slots would be for the support of different time granularities?

Reply:
std::optional would allow a bit of type-safe so that you do not accidentally use TimesliceSlot::INVALID as an actual index. But yeah, maybe for later.

Multiple Timeslice slots could be useful for things which populate the index at once, rather than waiting to be called many times (say for example if we use the same mechanism to read from file and we know in advance we will need N time slots). That said, again this is probably for later.

Reply:
In that sense - indeed, though it is not related to these fixes, so I would leave it as it is here.